java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/InputStream.java

```diff
@@ -1620,29 +1611,6 @@ protected String readTypeId(boolean isIndex) {
             }
         }
 
-        protected Class<?> resolveClass(String typeId) {
-            Class<?> cls = null;
-            if (_typeIdCache == null) {
-                _typeIdCache = new HashMap<>(); // Lazy initialization.
-            } else {
-                cls = _typeIdCache.get(typeId);
-            }
-
-            if (cls == EncapsDecoder.class) {// Marker for non-existent class.
```

Comment:
I did not (and could not) keep this "not found" caching. See comments in DefaultPackageSliceLoader.

I guess it could be meaningful (performance wise) if you get of unknown  derived classes that you then slice to a known base. In this unlikely situation, you could install your own SliceLoader that somehow optimizes "not found".